### ISSUE TITLE
OpenSplice with Connext tests fail on OS X

### ISSUE THREAD

**Issue by wjwwood** [ros2/system_tests#8]
This results in 24 failing tests. It seems to be related to discovery so it affects all of the pairs (connext publishing with opensplice subscribing and vise versa). I can also reproduce this locally by running this:

```
% ./install/bin/listener__rmw_connext_cpp

```

And then this in a separate terminal:

```
% OSPL_VERBOSITY=2 OSPL_INFOFILE="<stdout>" OSPL_ERRORFILE="<stderr>" ./install/bin/talker__rmw_opensplice_cpp
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:42 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897802.113387000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:42 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897802.113441000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:43 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897803.117932000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:43 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897803.118007000
Publishing: 'Hello World: 1'
Publishing: 'Hello World: 2'
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:44 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897804.121762000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:44 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897804.121831000
Publishing: 'Hello World: 3'
Publishing: 'Hello World: 4'
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:45 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897805.126194000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:45 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897805.126253000
Publishing: 'Hello World: 5'
Publishing: 'Hello World: 6'
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:46 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897806.127074000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:46 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897806.127108000
Publishing: 'Hello World: 7'
Publishing: 'Hello World: 8'
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:47 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897807.129180000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:47 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897807.129228000
Publishing: 'Hello World: 9'
Publishing: 'Hello World: 10'
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:48 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897808.132293000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 17:56:48 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : talker__rmw_opensplice_cpp <17916>
Thread      : dq.builtins 10fa2d000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433897808.132369000
Publishing: 'Hello World: 11'
Publishing: 'Hello World: 12'
Publishing: 'Hello World: 13'
Publishing: 'Hello World: 14'
Publishing: 'Hello World: 15'
Publishing: 'Hello World: 16'
Publishing: 'Hello World: 17'
Publishing: 'Hello World: 18'
Publishing: 'Hello World: 19'
Publishing: 'Hello World: 20'
^Csignal_handler(2)
```

Eventually the warnings settle out, but they never exchange messages. This seems new to me, at one point communication between the two was working on OS X.

**Comment by wjwwood** [comment]
Here is the output with the roles swapped (opensplice subscribing):

```
% OSPL_VERBOSITY=2 OSPL_INFOFILE="<stdout>" OSPL_ERRORFILE="<stderr>" ./install/bin/listener__rmw_opensplice_cpp
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:49 PDT 2015
Description : malformed packet received from vendor 1.1 state parse:shortmsg <52545053 02010101 5344444e 474e4950 @0x14 > (note: maybe partially bswap'd)

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : recv 10e71e000
Internals   : V6.4.140407OSS///ddsi2/q_receive.c/2403/0/1433898049.863989000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:49 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898049.864096000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:49 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898049.864124000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:50 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898050.868156000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:50 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898050.868205000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:51 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898051.871808000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:51 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898051.872228000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:52 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898052.873669000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:52 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898052.873741000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:53 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898053.875226000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:53 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898053.875369000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:54 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898054.877549000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:54 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898054.877651000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:55 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898055.877892000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:00:55 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898055.877938000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:01:00 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898060.345178000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:01:00 PDT 2015
Description : malformed packet received from vendor 1.1 state header <52545058 02010101 ac1702d4 00004611 00000001 09010800 2e8c7755 5abc1f57> (note: maybe partially bswap'd)

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : recv 10e71e000
Internals   : V6.4.140407OSS///ddsi2/q_receive.c/2297/0/1433898060.345457000
========================================================================================
Report      : WARNING
Date        : Tue Jun 09 18:01:00 PDT 2015
Description : SPDP (vendor 1.1): invalid qos/parameters

Node        : farl
Process     : listener__rmw_opensplice_cpp <17947>
Thread      : dq.builtins 10e655000
Internals   : V6.4.140407OSS///ddsi2/q_ddsi_discovery.c/683/0/1433898060.345526000
^Csignal_handler(2)
```

There's no erroneous output from the Connext talker.

**Comment by wjwwood** [comment]
The same exact things happen with Connext dynamic as well.

**Comment by wjwwood** [comment]
OpenSplice to OpenSplice seems to work fine.

**Comment by wjwwood** [comment]
Closed by #69 and the related pr's.
